proposals/NNNN-Add-function-to-transmit-audio-data.md

```diff
@@ -0,0 +1,973 @@
+                            }
+                            reference.decoder.releaseOutputBuffer(outputBuffersArrayIndex, false);
+                        }
+                    } else if (outputBuffersArrayIndex == MediaCodec.INFO_OUTPUT_FORMAT_CHANGED) {
+                        MediaFormat newFormat = reference.decoder.getOutputFormat();
+                        reference.onOutputFormatChanged(newFormat);
+                    }
+                } catch (Exception e) {
+                    Log.d(TAG, "DecoderRunnable Exception OutputBuffer :" + e);
+                    return true;
+                }
+            }
+            if (outputBufferInfo.flags == MediaCodec.BUFFER_FLAG_END_OF_STREAM) {
+                reference.listener.onAudioDataAvailable(null,outputBufferInfo.flags);
+                if (reference.listener != null) {
+                    reference.listener.onDecoderFinish(true);
+                }
+                reference.stop();
+                return true;
+            }
+            return false;
+        }
+    }
+```
+####BaseAudioDecoder.java file
+#####Output buffer
+```java
+	protected ArrayList<SampleBuffer> onOutputBufferAvailable(@NonNull 			ByteBuffer outputBuffer) {
+		double outputPresentationTimeUs = lastOutputPresentationTimeUs;
+		double outputDurationPerSampleUs = 1000000.0 / (double)outputSampleRate;
+
+		double targetPresentationTimeUs = lastTargetPresentationTimeUs;
+		double targetDurationPerSampleUs = 1000000.0 / (double)targetSampleRate;
+
+		// wrap the output buffer to make it provide audio samples
+		SampleBuffer outputSampleBuffer = SampleBuffer.wrap(outputBuffer, outputSampleType, outputChannelCount, (long)outputPresentationTimeUs);
+		outputSampleBuffer.position(0);
+		SampleBuffer targetSampleBuffer = null;
+		Double sample;
+		ArrayList<SampleBuffer> targetSampleBufferList = new ArrayList<>();
+		do {
+			sample = sampleAtTargetTime(lastOutputSample, outputSampleBuffer, outputPresentationTimeUs, outputDurationPerSampleUs, targetPresentationTimeUs);
+			if(targetSampleBuffer == null){
+				//1 second worth of data
+				targetSampleBuffer = SampleBuffer.allocate((targetSampleRate/1000) * 1024, targetSampleType, ByteOrder.LITTLE_ENDIAN, (long)targetPresentationTimeUs);
+			}
+			if (sample != null) {
+				targetSampleBuffer.put(sample);
+				targetPresentationTimeUs += targetDurationPerSampleUs;
+			}
+			if(sample == null || targetSampleBuffer.getByteBuffer().remaining() < 2){
+				targetSampleBuffer.limit(targetSampleBuffer.position());
+				targetSampleBuffer.position(0);
+				targetSampleBufferList.add(targetSampleBuffer);
+				targetSampleBuffer = null;
+			}
+		} while (sample != null);
+
+		lastTargetPresentationTimeUs = targetPresentationTimeUs;
+		lastOutputPresentationTimeUs += outputSampleBuffer.limit() * outputDurationPerSampleUs;
+		lastOutputSample = outputSampleBuffer.get(outputSampleBuffer.limit() - 1);
+		return targetSampleBufferList;
+}
+```
+
+## Potential downsides
```

Comment:
I'd recommend including some information in this section, especially given the amount of code changes proposed in the solution.

Reply:
I’ve added the information needed.